src/mscorlib/src/System/Collections/Generic/Comparer.cs

```diff
@@ -77,9 +108,26 @@ private static Comparer<T> CreateComparer()
             return 0;
         }
     }
-
+    
+    // Helper class: Represents Comparer subclasses that do not have any fields.
+    // Their identity can be calculated as a function of their type.
     [Serializable]
-    internal class GenericComparer<T> : Comparer<T> where T: IComparable<T>
+    internal abstract class StatelessComparer<T> : Comparer<T>
```

Comment:
@jkotas OK, got it. I'll move out all of the shared logic from (and delete) the `EnumComparer` class as well.